### PR TITLE
Send magnet links directly to rtorrent.

### PR DESCRIPTION
**Pull request by sdrycroft**
Send magnet links directly to rtorrent, rather than trying to download the torrent file and sending the data.

### CODE REVIEW

**Comment by sdrycroft** [comment]
Third time lucky, apologies for the previous two pull requests.

**Comment by RuudBurger** [comment]
Thanks for the PR, but you let it return the wrong data like this. Also ignoring some options the user has set (pause download etc). Please check my comments

File: couchpotato/core/downloaders/rtorrent_.py [hunk 1]
```diff
@@ -156,12 +156,13 @@ def download(self, data = None, media = None, filedata = None):
...
-                return False
-
-            data['protocol'] = 'torrent'
+            magnet_url = data.get('url')
+            # Pull out the ID of the torrent
+            urn_btih_position = magnet_url.index('urn:btih:')
+            torrent_hash = magnet_url[urn_btih_position + 9:urn_btih_position+49]
+            # Send magnet URL directly to rTorrent
+            self.rt.load_torrent_simple(magnet_url, 'url', start=True)
+            return torrent_hash
```
Comment (RuudBurger) [inline couchpotato/core/downloaders/rtorrent_.py]: It should return a download id of the downloader. like below `self.downloadReturnId(torrent_hash)`

File: couchpotato/core/downloaders/rtorrent_.py [hunk 2]
```diff
@@ -156,12 +156,13 @@ def download(self, data = None, media = None, filedata = None):
...
-            if filedata is False:
-                return False
-
-            data['protocol'] = 'torrent'
+            magnet_url = data.get('url')
+            # Pull out the ID of the torrent
+            urn_btih_position = magnet_url.index('urn:btih:')
+            torrent_hash = magnet_url[urn_btih_position + 9:urn_btih_position+49]
+            # Send magnet URL directly to rTorrent
+            self.rt.load_torrent_simple(magnet_url, 'url', start=True)
```
Comment (RuudBurger) [inline couchpotato/core/downloaders/rtorrent_.py]: This should be below and in an if/else statement with the `torrent = self.rt.load_torrent(filedata, verify_retries=10)`

Returning the torrent request, check if it download succeeded and set label/directory/paused

**Pull request closed without merging**